### PR TITLE
MM-59440 Add unit tests for actions/remote/channel

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Add unit tests for actions/remote/channel. Coverage increases from 5.01% to 88.24%.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-59440

#### Checklist

- [x] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.
- [ ] Have run E2E tests by adding label `E2E iOS tests for PR`.

#### Device Information
This PR was tested on: n/a

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/actions/remote/channel.ts [hunk 1]
```diff
@@ -798,7 +798,7 @@ export async function createDirectChannel(serverUrl: string, userId: string, dis
             const config = await getConfig(database);
             const teammateDisplayNameSetting = getTeammateNameDisplaySetting(preferences || [], config.LockTeammateNameDisplay, config.TeammateNameDisplay, license);
             const {directChannels, users} = await fetchMissingDirectChannelsInfo(serverUrl, [created], currentUser.locale, teammateDisplayNameSetting, currentUser.id, true);
-            created.display_name = directChannels?.[0].display_name || created.display_name;
+            created.display_name = (directChannels?.length && directChannels?.[0].display_name) || created.display_name;
```
Comment (enahum) [inline app/actions/remote/channel.ts:801]: Why this change? What did you find?

File: app/actions/remote/channel.test.ts [hunk 2]
```diff
@@ -0,0 +1,762 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable max-lines */
+
+import {createIntl} from 'react-intl';
+
+import {DeepLink} from '@app/constants';
+import {SYSTEM_IDENTIFIERS} from '@app/constants/database';
```
Comment (enahum) [inline app/actions/remote/channel.test.ts:9]:
```suggestion
import {DeepLink} from '@constants';
import {SYSTEM_IDENTIFIERS} from '@constants/database';
```

File: app/actions/remote/channel.test.ts [hunk 3]
```diff
@@ -0,0 +1,762 @@
...
+jest.mock('@queries/app/servers', () => {
+    const original = jest.requireActual('@queries/app/servers');
+    mockGetActiveServer = jest.fn(() => false);
+    return {
+        ...original,
+        getActiveServer: mockGetActiveServer,
+    };
+});
+
+const mockClient = {
```
Comment (enahum) [inline app/actions/remote/channel.test.ts:92]: I wonder if it would be better to use nock as in https://github.com/mattermost/mattermost-mobile/blob/main/test/test_helper.ts#L527

**Review by enahum — APPROVED** [review]
None blocking but for consideration

File: app/actions/remote/channel.ts (hunk 1 above)
Comment (jwilander) [reply app/actions/remote/channel.ts:801]: It's possible to have `directChannels` be an empty array which would crash here. It's possible that this can't happen through normal usage of the app but in mocked environments like the unit tests it was happening. So I updated this to be resilient to that

File: app/actions/remote/channel.test.ts (hunk 3 above)
Comment (jwilander) [reply app/actions/remote/channel.test.ts:92]: I was following the example set here: https://github.com/mattermost/mattermost-mobile/blob/main/app/products/calls/actions/calls.test.ts#L41

0/5 whether to use nock instead. Since it's a unit test I think it's fine to mock at a shallower level like I've done. Unless there's a strong opinion to change it to nock then I'll leave it as is

Comment (enahum) [reply app/actions/remote/channel.test.ts:92]: Is ok, approved already so no strong opinion

**Review by yasserfaraazkhan — APPROVED** [review]
🎉 this was a big one.
